src/index.test.ts

```diff
@@ -1356,6 +1357,46 @@ test('Should pass custom headers', async () => {
     });
 });
 
+test('Should add X-UNLEASH headers', async () => {
+    fetchMock.mockResponses(
+        [JSON.stringify(data), { status: 200 }],
+        [JSON.stringify(data), { status: 200 }]
+    );
+    const appName = 'unleash-client-test';
+    const config: IConfig = {
+        url: 'http://localhost/test',
+        clientKey: 'some123key',
+        appName,
+    };
+    const client = new UnleashClient(config);
+    await client.start();
+
+    const featureRequest = getTypeSafeRequest(fetchMock, 0);
+
+    client.isEnabled('count-metrics');
+    jest.advanceTimersByTime(2001);
+
+    const metricsRequest = getTypeSafeRequest(fetchMock, 1);
+
+    const uuidFormat =
+        /^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$/i;
+
+    const expectedHeaders = {
+        'x-unleash-sdk': `unleash-js@${sdkVersion}`,
+        'x-unleash-connection-id': expect.stringMatching(uuidFormat),
+        'x-unleash-appname': appName,
+    };
+
+    const getConnectionId = (request: any) =>
+        JSON.parse(JSON.stringify(request.headers))['x-unleash-connection-id'];
```

Comment:
would `const getConnectionId = (request: any) => request.headers['x-unleash-connection-id'];` work?

Reply:
I hoped it would, but no such luck when I tried yesterday. I'll try again.

Reply:
Huh, it worked in the function, but not when applied to the header directly. My guess is that because we say it's `any`, then we lose any checking that would prevent you from getting the header.

It didn't work yesterday because trying to do `featureRequest.headers['x-unleash-connection-id']` gives you this error:

```
Element implicitly has an 'any' type because expression of type '"x-unleash-connection-id"' can't be used to index type 'HeadersInit'.
      Property 'x-unleash-connection-id' does not exist on type 'HeadersInit'.
```